datacube/drivers/postgis/_connections.py

```diff
@@ -233,11 +233,25 @@ def create_spatial_index(self, crs: CRS) -> Optional[Type[SpatialIndex]]:
                 _LOG.warning("Could not dynamically model an index for CRS %s", crs._str)
                 return None
             ensure_spindex(self._engine, spidx)
-            self.spindexes[crs] = spidx
+            self._refresh_spindexes()
         return spidx
 
+    def drop_spatial_index(self, crs: CRS) -> bool:
+        """
+        Create a spatial index across the database, for the named CRS.
+
+        :param crs:
+        :return:
+        """
+        spidx = self.spindexes.get(crs)
+        if spidx is None:
+            return False
+        result = drop_spindex(self._engine, spidx)
+        self._refresh_spindexes()
+        return result
+
     def spatial_index(self, crs: CRS) -> Optional[Type[SpatialIndex]]:
-        return self.spindexes.get(crs)
+        return self.spindexes.get(normalise_crs(crs))
 
     def spatial_indexes(self, refresh=False) -> Iterable[CRS]:
```

Comment:
This might be worth renaming, since the return type is not `SpatialIndex`

Reply:
The return type is an ORM class that is guaranteed to be a subclass of SpatialIndex...

How about `spatial_index_orm(crs)`?

Reply:
GitHub comments once again making it unclear what I'm commenting on... This comment was in reference to `spatial_indexes`, which returns CRSs. I have no issues with the name of `spatial_index`.

Reply:
Ah that does make more sense.